### PR TITLE
AEA-1301 Add separation of decision maker

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR brings separation to the DM handler files. We have two files `default` which is used for projects by default. This DM does not do any validation, it just signs any message/transaction it receives. `gop` (for `goals, ownership and preferences`) is used in tac only atm. Over time we can add more DMs.

## Fixes

- #2223 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/decision_maker/default.py [hunk 1]
```diff
@@ -19,22 +19,12 @@
 
 """This module contains the decision maker class."""
 
-import copy
-import logging
```
Comment (DavidMinarsch) [inline aea/decision_maker/default.py:23]: Henceforth the default decision maker just accepts signing messages and processes them. There is no further checks.

File: aea/decision_maker/gop.py [hunk 2]
```diff
@@ -0,0 +1,811 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
```
Comment (DavidMinarsch) [inline aea/decision_maker/gop.py:7]: At the moment this is just used for TAC, so makes no sense to use it as default.

File: packages/fetchai/agents/tac_participant/aea-config.yaml [hunk 3]
```diff
@@ -33,3 +33,6 @@ default_routing:
   fetchai/ledger_api:0.10.0: fetchai/ledger:0.13.0
   fetchai/oef_search:0.13.0: fetchai/soef:0.16.0
 connection_private_key_paths: {}
+decision_maker_handler:
+  dotted_path: aea.decision_maker.gop:DecisionMakerHandler
```
Comment (DavidMinarsch) [inline packages/fetchai/agents/tac_participant/aea-config.yaml:37]: This is how `aea` specific DMs can be modified (default points to `aea.decision_maker.default:DecisionMakerHandler`)

File: tests/test_packages/test_skills_integration/test_tac.py [hunk 4]
```diff
@@ -147,6 +147,12 @@ def test_tac(self):
             self.set_config("agent.default_ledger", FETCHAI)
             setting_path = "agent.default_routing"
             self.nested_set_config(setting_path, default_routing)
+            data = {
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills_integration/test_tac.py:150]: Need to also update docs (note to self)

File: aea/aea_builder.py [hunk 5]
```diff
@@ -441,7 +442,10 @@ def set_decision_maker_handler(
         dotted_path, class_name = decision_maker_handler_dotted_path.split(
             DOTTED_PATH_MODULE_ELEMENT_SEPARATOR
         )
-        module = load_module(dotted_path, file_path)
+        if file_path is None:
+            module = import_module(dotted_path)
+        else:
+            module = load_module(dotted_path, file_path)
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:448]: Need to add try/except

File: tests/test_decision_maker/test_default.py [hunk 6]
```diff
@@ -631,5 +464,8 @@ def test_handle_messages_from_two_dialogues_same_agent(self):
     @classmethod
     def teardown(cls):
         """Tear the tests down."""
-        cls._unpatch_logger()
         cls.decision_maker.stop()
+
+
+class TestDecisionMaker(BaseTestDecisionMaker):
```
Comment (5A11) [inline tests/test_decision_maker/test_default.py:470]: what is this class's purpose?

Comment (DavidMinarsch) [reply tests/test_decision_maker/test_default.py:470]: It runs the test `Test`

File: aea/aea_builder.py (hunk 5 above)
Comment (marcofavorito) [reply aea/aea_builder.py:448]: I think somewhere we need to explain that `file_path` overwrites the behaviour of `dotted_path`

**Review by marcofavorito — APPROVED** [review]
LGTM, just a comment on priority of `file_path` vs `dotted_path`

File: aea/aea_builder.py (hunk 5 above)
Comment (DavidMinarsch) [reply aea/aea_builder.py:448]: It does not overwrite..

**Pull request merged**